engine/engine_test.go

```diff
@@ -2061,6 +2061,16 @@ func TestInstantQuery(t *testing.T) {
 						http_requests_total{pod="nginx-2"} 1`,
 			query: "max(http_requests_total) by (pod)",
 		},
+		{
+			name: "max by with NaN",
```

Comment:
Do we need to fix this for other aggregations too, like `sum`, `min` etc?

Also can we have a test for aggregations without grouping?

Reply:
I added couple tests; the other cases ( no grouping, NaN in sum and min ) seem to be ok!